### PR TITLE
(CTH-390) Add PXP agent init scripts

### PR DESCRIPTION
**Pull request by geoffnichols**
This commit adds init scripts and service
configuration files for initialization of
the PXP agent service.

These are derived as wholly as possible from
the facilities currently used for Puppet in
the AIO puppet-agent.

### CODE REVIEW

File: ext/osx/pxp-agent.plist [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+<dict>
+        <key>EnvironmentVariables</key>
+        <dict>
+                <key>PATH</key>
+                <string>/opt/puppetlabs/puppet/bin:/usr/bin:/bin:/usr/sbin:/sbin</string>
+                <key>LANG</key>
+                <string>en_US.UTF-8</string>
+        </dict>
+        <key>Label</key>
+        <string>pxp-agent</string>
```
Comment (heathseals) [inline ext/osx/pxp-agent.plist]: @stahnma @kylog we decided that having it fully qualified was the correct way to do it ( and would require managing it with the full name ), but there would be additional work elsewhere.

Comment (kylog) [inline ext/osx/pxp-agent.plist]: @heathseals does that mean:
a) we're going to leave it as-is in this PR, i.e. not fully qualified
OR
b) we're going to change it to be fully qualified, and the additional work is ticketed
OR
something else?

File: ext/debian/pxp-agent.init [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+### END INIT INFO
+
+PATH=/opt/puppetlabs/puppet/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
+DAEMON=/opt/puppetlabs/puppet/bin/pxp-agent
+DAEMON_OPTS="--daemonize "
+PROCNAME="pxp-agent"
+DESC="PXP agent"
+PIDFILE="/var/run/puppetlabs/${PROCNAME}.pid"
+
+test -x $DAEMON || exit 0
```
Comment (haus) [inline ext/debian/pxp-agent.init]: I think this should exit 5 according to lsb.

File: ext/debian/pxp-agent.init [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    start-stop-daemon --stop --quiet --signal HUP --pidfile $PIDFILE --name $PROCNAME
+}
+
+start_pxp_agent() {
+    start-stop-daemon --start --quiet --pidfile $PIDFILE --startas $DAEMON $DAEMON_OPTS $PXP_AGENT_EXTRA_OPTS
+}
+
+stop_pxp_agent() {
+    start-stop-daemon --stop --retry TERM/10/KILL/5 --quiet --oknodo --pidfile $PIDFILE --name $PROCNAME
+    rm -f "$PIDFILE"
```
Comment (haus) [inline ext/debian/pxp-agent.init]: do we always want the pidfile blown away or only on success?

File: ext/debian/pxp-agent.init [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+restart_pxp_agent() {
+    log_begin_msg "Restarting $DESC"
+    stop_pxp_agent
+    start_pxp_agent
+    log_end_msg $?
```
Comment (haus) [inline ext/debian/pxp-agent.init]: We don't ever check to see if the stop was successful here. Do we care? Should the restart stop if the stop failed?

**Comment by haus** [comment]
@geoffnichols the variables used and cases vary greatly for the svsv scripts (deb, el, sles). These would be easier to follow if they used the same variables (DAEMON vs pxpagentd for example). Also the sysconfig/default files should probably have the same options exposed, which currently isn't the case.

File: ext/redhat/pxp-agent.init [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+export PATH
+
+[ -f /etc/sysconfig/pxp-agent ] && . /etc/sysconfig/pxp-agent
+lockfile=/var/lock/subsys/pxp-agent
+piddir=/var/run/puppetlabs
+pidfile=${piddir}/pxp-agent.pid
+pxp_agentd=/opt/puppetlabs/puppet/bin/pxp-agent
+pid=$(cat $pidfile 2> /dev/null)
+RETVAL=0
+
```
Comment (haus) [inline ext/redhat/pxp-agent.init]: Any reason we aren't checkng to see if pxp_agentd is executable here but we did in the debian? Same question for sles.

File: ext/debian/pxp-agent.default [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# Defaults for pxp-agent - sourced by /etc/init.d/pxp-agent
+
+# Startup options
+#DAEMON_OPTS=""
```
Comment (haus) [inline ext/debian/pxp-agent.default]: DAEMON_OPTS is called PXP_AGENT_OPTS in the el init script. We should probably pick one.

**Comment by geoffnichols** [comment]
Since the puppetlabs/pxp-agent repo was made public after this PR was opened, it looks like I need to close this PR and open a new one (with a new fork of puppetlabs/pxp-agent). 

Pleas see https://github.com/puppetlabs/pxp-agent/pull/112.

**Pull request closed without merging**